aggregator/src/aggregator.rs

```diff
@@ -1806,9 +1806,39 @@ impl VdafOps {
                                     Ok::<_, datastore::Error>((batch_identifier, batch))
                                 }
                             }),
-                        )
+                        ),
+                        try_join_all(
+                            Q::batch_identifiers_for_collection_identifier(
+                                &task,
+                                &collection_identifier
```

Comment:
One of these calls is retrieving `batches`, the other is checking a property of the `client_reports` table.

`Q::batch_identifiers_for_collection_identifier` is an iterator, and doesn't perform network IO or any other blocking operation -- concretely, time-interval tasks just generate the sequence of minimal intervals included in the collection interval, while fixed-size tasks just give back the batch identifier directly. I don't think it's doing enough work to make reusing it valuable, especially since I think we'd either need to (a) collect the batch identifiers into a data structure [using a bit of memory] or (b) map the batch identifiers into a single future that retrieves both the `batches` & `client_reports` information [and have to write some awkward code to split them back out from the result of the `try_join_all`].

(also, this is pedantic, but when evaluating the effective number of network round trips, my model so far has been that concurrent requests -- e.g. in the same `try_join!`/`try_join_all` -- are a single logical round trip. this is because, due to pipelining, we can send all the requests to Postgres at once then receive the responses -- so these requests are already in one "effective" DB round trip. while this model isn't perfect, e.g. I'd bet there is a limit to the number of outstanding pipelined requests that may come into play, we haven't identified performance issues due to it yet.)